tests/func/params/test_show.py

```diff
@@ -44,6 +44,23 @@ def test_show_toml(tmp_dir, dvc):
     }
 
 
+def test_preserve_comments(tmp_dir, dvc):
```

Comment:
Can you create a new test file `tests/unit/utils/serialize/test_toml.py` and move all of these 3 tests there, please? They are not related to `test_show` or `test_params`, but lower level utilities.

Regarding the test, please try using (preferably) `modify_toml()` or `dump_toml` instead of `_dump`.

Reply:
I'm struggling to get `modify_toml()` working

Reply:
yeah, that's a bug in `modify_data`, which returns an empty dict if the file does not exist.

https://github.com/iterative/dvc/blob/634894d3b1205f0006a8404d9914e7916a877355/dvc/utils/serialize/_common.py#L87

We can ignore this, as `modify_toml` is about updating contents (raw/dict) to existing content (special dict). It's not the other way around. You can simply test as follows:
```python
def test_preserve_comments(tmp_dir):
    from dvc.utils.serialize import _toml

    contents_fmt = """\
#A Title
[foo]
bar = {} # meaning of life
baz = [1, 2]
"""
    tmp_dir.gen("params.toml", contents_fmt.format("42"))

    with _toml.modify_toml("params.toml") as d:
        d["foo"]["bar"] //= 2
    assert (tmp_dir / "params.toml").read_text() == contents_fmt.format("21")

```

Reply:
@syntapy Not sure how you are trying to test `modify_toml`. It works as expected for me when trying in how I believe should be used:

```
$ cat foo.toml 
```

```toml
# A Title 
[foo]
bar = 42# meaning of life
baz = [1, 2]
```

```python
$ python
>>> from dvc.utils.serialize._toml import modify_toml
>>> with modify_toml("foo.toml") as data:
...     data["foo"]["bar"] = 2
... 
```

```
$ cat foo.toml
```
```toml
# A Title 
[foo]
bar = 2# meaning of life
baz = [1, 2]
```

Reply:
@daavoo, what @syntapy was saying is that it fails when you try to update the content of `foo.toml` to a new file that does not exist yet. That scenario, however is not something that we support, or intend to support with `modify_data`/`modify_toml`.

i.e

```python
from dvc.utils.serialize._toml import modify_toml

with open("foo.toml") as f:
  content = parse_toml_for_update(f.read(), "foo.toml")
with modify_toml("new_file_that_does_not_exist_yet.toml") as d:
  d.update(content)
```

The top level comment is not preserved.

Reply:
> That scenario, however is not something that we support, or intend to support with modify_data/modify_toml.

My point was that the test was not testing what we use `modify_toml` for